src/notification/index.js

```diff
@@ -247,17 +234,22 @@ export class NotificationListener {
       this.listenAndroid('notificationOpened', this.handleNotificationOpen);
       this.listenAndroid('remoteNotificationsRegistered', this.handleDeviceToken);
     } else {
-      // On iOS, `note` should be an IOSNotifications object. The notification
-      // data it returns from `getData` is unvalidated -- it comes almost
-      // straight off the wire from the server.
-      this.listenIOS('notificationOpened', (note: { getData(): JSONableDict }) => {
-        const data = fromAPNs(note.getData());
-        if (data) {
-          this.handleNotificationOpen(data);
+      this.listenIOS('notification', (notification: PushNotificationIOS) => {
+        // This is actually typed as ?Object (and so effectively `any`); but if
+        // present, it must be a JSONable dictionary. (See PushNotificationIOS.js and
+        // RCTPushNotificationManager.m in Libraries/PushNotificationIOS.)
+        const data: ?JSONableDict = notification.getData();
```

Comment:
> It might be simpler to appeal to what this method's job is: it's giving us the notification data, which was passed over APNs as JSON.

Would there still be reason to use `JSONableDict`, or would we have to just say `JSONable` on that justification?

Reply:
I think `JSONableDict` is still fine. The APNs protocol has `data` as a blob which is supposed to be a JSON dictionary.

Though if we can say `JSONable` here and that doesn't cause complications anywhere downstream, then that'd be cleaner still.